lib/pages/settings/view/settings_page.dart

```diff
@@ -18,8 +23,13 @@ class SettingsPage extends StatefulWidget {
 }
 
 class SettingsPageState extends State<SettingsPage> {
+  User user;
+  AuthProvider authProvider;
+
   @override
   Widget build(BuildContext context) {
+    authProvider = Provider.of(context);
```

Comment:
I'm surprised this isn't giving you an error/warning. You need to specify the type:
```suggestion
    authProvider = Provider.of<AuthProvider>(context);
```